controllers/object_controls.go

```diff
@@ -1238,6 +1238,23 @@ func TransformToolkit(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n
 
 	// configure runtime
 	runtime := n.runtime.String()
+	err = TransformDaemonsetForRuntime(obj, config, runtime)
+	if err != nil {
+		return fmt.Errorf("error transforming toolkit daemonset : %v", err)
+	}
+
+	// Update CRI-O hooks path to use default path for non OCP cases
+	if n.openshift == "" && n.runtime == gpuv1.CRIO {
+		for index, volume := range obj.Spec.Template.Spec.Volumes {
+			if volume.Name == "crio-hooks" {
+				obj.Spec.Template.Spec.Volumes[index].HostPath.Path = "/usr/share/containers/oci/hooks.d"
+			}
+		}
+	}
+	return nil
+}
+
+func TransformDaemonsetForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, runtime string) error {
```

Comment:
Instead of hardcoding `obj.Spec.Template.Spec.Containers[0]` throughout this function to refer to the main container, let's pass the `mainContainerName` as input to this function and search for it
```suggestion
func TransformDaemonsetForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, runtime string, mainContainerName string) error {
    var mainContainer *corev1.Container
    for i, ctr := range obj.Spec.Template.Spec.Containers {
        if ctr.Name == mainContainerName {
            mainContainer = &obj.Spec.Template.Spec.Containers[i]
            break 
        }
    }
    if mainContainer == nil {
        return fmt.Errorf("failed to find main container %q", mainContainerName)
    }
```

Reply:
Makes sense. Fixed it.